R/run_tmb.R

```diff
@@ -839,11 +841,11 @@ g3_tmb_adfun <- function(cpp_code,
     base_name <- paste0('g3_tmb_', digest::sha1(cpp_code))
     cpp_path <- paste0(file.path(work_dir, base_name), '.cpp')
     so_path <- TMB::dynlib(file.path(work_dir, base_name))
-
+    
     # If not loaded yet, compile & load
     if (!any(vapply(getLoadedDLLs(), function (x) x[['path']] == so_path, logical(1)))) {
-        writeLines(cpp_code, con = cpp_path)
-
+        writeLines(cpp_code, con = gsub("\\\\", "/", cpp_path))
```

Comment:
Is this necessary? Invariably ``work_dir`` is your tempdir, and R must be able to write to that without path shenanigans.